### PR TITLE
#159987424 Comments should be able to track edit history

### PR DESCRIPTION
**Pull request by jamesenejo**
#### What does this PR do?
Adds the functionality of tracking histories of comments and replies

#### Description of Task to be completed?
Have the following endpoints working

- POST /api/v1/articles/:articleId/comments/:commentId/reply
- PUT /api/v1/articles/:articleId/comments/:commentId
- PUT /api/v1/articles/:articleId/comments/:commentId/reply/:replyId

#### How should this be manually tested?
- After cloning the repo, cd into it and run 'npm run start_dev'
- Using postman, you can test the endpoint

#### What are the relevant pivotal tracker stories?
#159987424

### CODE REVIEW

File: test/commentupdate.spec.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        .send({
+          content: 'Your post was not inspiring.'
+        })
+        .end((err, res) => {
+          res.body.status.should.equal('fail');
+          res.body.data.message.should.equal('Invalid commentId');
+          done();
+        });
+    });
+    it('should fail if commentId is invalid', (done) => {
```
Comment (tomiadebanjo) [inline test/commentupdate.spec.js]: Nice Work James  👍. You have a duplicate test description for two different tests on line 43 and line 57. You could easily make corrections to the test.

Comment (tomiadebanjo) [reply test/commentupdate.spec.js]: You also have the same description for two tests on line 99 and 113.

Comment (jamesenejo) [reply test/commentupdate.spec.js]: That's an eagle eye look Tomi. Thank you for noticing. I would make changes to it in my next branch.

File: server/controllers/likeController.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import models from '../models/index';
-import liker from '../helpers/liker';
+import commentHelpers from '../helpers/commentHelpers';
 
 const { CommentLikes, ReplyLikes } = models;
+const { liker, } = commentHelpers;
```
Comment (iAmao) [inline server/controllers/likeController.js]: remove comma

File: server/helpers/commentHelpers.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+const commentHelpers = {
+  liker: (model, response, likeData, type) => {
+    model
+      .findOrCreate({ where: likeData, defaults: { liked: true } })
+      .spread((result, created) => {
+        const { liked } = result.dataValues;
+
+        // When user is liking for the first time
+        if (liked && created) {
```
Comment (iAmao) [inline server/helpers/commentHelpers.js]: we can just do if created here

File: server/helpers/commentHelpers.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+          return response.status(201).jsend.success({
+            message: `${type} liked`
+          });
+        }
+
+        // The two if blocks below toggle the like functionality
+        // this runs when a user unlikes
+        if (liked) {
+          return model
```
Comment (iAmao) [inline server/helpers/commentHelpers.js]: Let's try something like:
```
return model
          .update({ liked: !liked }, { where: likeData })
          .then(() => response.status(200).jsend.success({
            message: `${type} ${liked ? 'unliked' : 'liked'}`
          }));
```

Comment (iAmao) [reply server/helpers/commentHelpers.js]: So we can get rid of the if statements

File: server/middleware/commentValidator.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    // Check the passed body for required properties
+    const { valid, invalidMessages } = checkProps(req.body, 'id', 'type');
+
+    if (!valid) {
+      return res.status(400).jsend.fail({
+        messages: invalidMessages
+      });
+    }
+
+    // Validate the email address provided
```
Comment (iAmao) [inline server/middleware/commentValidator.js]: email?

File: server/middleware/commentValidator.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    if (!numberRegex.test(req.body.id)) {
+      status = 'fail';
+      messages.push('Invalid id');
+    }
+    // Validate the password provided
```
Comment (iAmao) [inline server/middleware/commentValidator.js]: update these comments

File: server/middleware/commentValidator.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    if ((req.body.type !== 'comment') && (req.body.type !== 'reply')) {
```
Comment (iAmao) [inline server/middleware/commentValidator.js]: Let's try this `if (!['comment', 'reply'].includes(req.body.type)) {`

**Review by iAmao — CHANGES_REQUESTED** [review]

**Pull request merged**